backend/src/tables/tables.service.ts

```diff
@@ -160,6 +163,38 @@ export class TablesService {
       TableSchema.parse(table),
     );
 
+    // Check the existing tables and generate a list of the new tables that will need to be created
+    const newTableNames = _.filter(_.keys(schemas), (tableName) => {
+      return !_.find(
+        parsedTables,
+        (table) => table.external_name === tableName,
+      );
+    });
+
+    // Use an LLM to generate table icons, colors and descriptions
+    const usedColors = _.map(parsedTables, (t) => t.color);
+    const usedIcons = _.map(parsedTables, (t) => t.icon);
```

Comment:
Lets move these maps inside the "if" block below, so they only get calculated if they are needed for the LLM